src/main.rs

```diff
@@ -145,6 +141,18 @@ fn get_pretend_executable() -> String {
                     .ok()
             }
         })
+        .or_else(|| {
+            // Try finding the executable from the command line
+            let path = env::args()
+                .nth(1)
+                .and_then(|exe| {
+                    which(exe).ok()
```

Comment:
This `which` call doesn't work with `set` because it's not a real executable but I think that's fine.

Reply:
Yes, it's fine. When absolute path is found, we run it as an executable otherwise it will be invoked through cmd.